dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -541,6 +541,7 @@ public Grid getAggregatedEventData(EventQueryParams params, Grid grid, int maxLi
       }
     } catch (BadSqlGrammarException ex) {
       log.info(AnalyticsUtils.ERR_MSG_TABLE_NOT_EXISTING, ex);
+      throw ex;
```

Comment:
I'm wondering if we couldn't just throw `new QueryRuntimeException(ErrorMessage errorMessage);`
In the error message, we could use the `ErrorCode` and define a message like:
`Bad SQL grammar. Error code "SQL_STATE_CODE"`. We could even have an Enum to hold the known SQL state codes (like the ones you have defined "42P01" and "42", and a generic one if a non-supported code is returned).

Note that QueryRuntimeException does not have a constructor for `ErrorMessage`, but I believe it could be added "_easily_".

In this way, we can return any SQL state of `BadSqlGrammarException`, and don't need to add any of the new classes below... The current flow will take care of everything. What do you think?